lib/queries/taxon_name/filter.rb

```diff
@@ -562,6 +573,38 @@ def leaves_facet
         leaves ? ::TaxonName.leaves : ::TaxonName.not_leaves
       end
 
+      # @return Scope
+      def latinized_facet
+        return nil if latinized.nil?
+
+        tnc = ::TaxonNameClassification.arel_table
+        if latinized == true
+          # Note the query here does not restrict to genus/species groups - a
+          # genus whose rank is changed to subfamily can retain its gender,
+          # e.g., and we want to include those here.
+          ::TaxonName.where(
+            ::TaxonNameClassification.where(
+              tnc[:taxon_name_id].eq(::TaxonName.arel_table[:id]).and(
+                tnc[:type].matches('%latinized%')
+              )
+            ).arel.exists
+          )
+        else
+          ::TaxonName
+            .where(
+              "rank_class ILIKE '%speciesgroup%' OR \
+               rank_class ILIKE '%genusgroup%'"
+            )
+            .where.not(
+              ::TaxonNameClassification.where(
+                tnc[:taxon_name_id].eq(::TaxonName.arel_table[:id]).and(
+                  tnc[:type].matches('%latinized%')
```

Comment:
If possible see if we can reference a constant that has the pertinent `type`s (`config/initializers/constants...`).  This lets us manage names == meaning nastiness in one place in the long run.  I.e. I *think* we should try to unify the general code pattern to avoid these matches.  We have this pattern in many places, it should(?) be improved(?) elsewhere too.

I'd not be too worried about such (again, tests are great) but since you're coming in a little more fresh to the nomenclatural side of things it might be an opportunity to get us all to be a little more consistent.

Reply:
> If possible see if we can reference a constant that has the pertinent `type`s (`config/initializers/constants...`). This lets us manage names == meaning nastiness in one place in the long run. I.e. I _think_ we should try to unify the general code pattern to avoid these matches.

That feels better to me, thanks for these pointers. To be honest I'm still frequently at the 'what's the right syntax for this query in rails (can I write this in rails)'/'I'm not sure how this rails/arel code works without to_sql'ing it' stage, so any thoughts on these things are always welcome.